### PR TITLE
Merge PaginatedList.pyi back to source

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: github/PaginatedList.py [hunk 1]
```diff
@@ -34,48 +34,61 @@
...
-    def _isBiggerThan(self, index):
+    def _isBiggerThan(self, index: int) -> bool:
         return len(self.__elements) > index or self._couldGrow()
 
     def __fetchToIndex(self, index):
         while len(self.__elements) <= index and self._couldGrow():
             self._grow()
 
-    def _grow(self):
+    def _grow(self) -> Any:
```
Comment (EnricoMi) [inline github/PaginatedList.py:79]: Shouldn' this be
```suggestion
    def _grow(self) -> Iterable[T]:
```

File: github/PaginatedList.py [hunk 2]
```diff
@@ -34,48 +34,61 @@
...
         newElements = self._fetchNextPage()
         self.__elements += newElements
         return newElements
 
     class _Slice:
-        def __init__(self, theList, theSlice):
+        def __init__(self, theList: "PaginatedListBase[T]", theSlice: slice) -> None:
```
Comment (EnricoMi) [inline github/PaginatedList.py:85]:
```suggestion
        def __init__(self, theList: "PaginatedListBase[T]", theSlice: slice):
```

File: github/PaginatedList.py [hunk 3]
```diff
@@ -237,7 +249,7 @@ def __parseLinkHeader(self, headers):
                 links[rel] = url
         return links
 
-    def get_page(self, page):
+    def get_page(self, page: int) -> List[Any]:
```
Comment (EnricoMi) [inline github/PaginatedList.py:252]:
```suggestion
    def get_page(self, page: int) -> List[T]:
```

File: github/PaginatedList.py [hunk 4]
```diff
@@ -34,48 +34,61 @@
...
-    def _isBiggerThan(self, index):
+    def _isBiggerThan(self, index: int) -> bool:
         return len(self.__elements) > index or self._couldGrow()
 
     def __fetchToIndex(self, index):
         while len(self.__elements) <= index and self._couldGrow():
             self._grow()
 
-    def _grow(self):
+    def _grow(self) -> Iterable[T]:
```
Comment (trim21) [inline github/PaginatedList.py:79]:
```suggestion
    def _grow(self) -> Iterator[T]:
```

File: github/Requester.py [hunk 5]
```diff
@@ -426,7 +426,7 @@ def requestJsonAndCheck(
         parameters: Optional[Dict[str, Any]] = None,
         headers: Optional[Dict[str, str]] = None,
         input: Optional[Any] = None,
-    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Any]]]:
```
Comment (EnricoMi) [inline github/Requester.py:429]: Why removing the `Dict` here?

File: github/Requester.py [hunk 6]
```diff
@@ -469,7 +469,7 @@ def __check(
         status: int,
         responseHeaders: Dict[str, Any],
         output: str,
-    ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
```
Comment (EnricoMi) [inline github/Requester.py:472]: Why removing the `Dict` here?

**Review by EnricoMi — CHANGES_REQUESTED** [review]

File: github/Requester.py (hunk 5 above)
Comment (trim21) [reply github/Requester.py:429]: It parse request response body, not necessarily dict

File: github/Requester.py (hunk 6 above)
Comment (trim21) [reply github/Requester.py:472]: Same above

Comment (EnricoMi) [reply github/Requester.py:472]: `requestJsonAndCheck` returns what `__check` returns, which is what `__structuredFromJson` returns, which calls into `json.loads(data)`, so should be `Optional[Dict[str, Any]]`.

Comment (trim21) [reply github/Requester.py:472]: Doesn't `json.loads(data)` actually return `Any`?

Comment (trim21) [reply github/Requester.py:472]: https://github.com/PyGithub/PyGithub/blob/b6258f4be9bee07beb0935a749289b152b20f7f0/github/PaginatedList.py#L201-L204

Comment (trim21) [reply github/Requester.py:472]: I'm 100% sure github API not only returns objects

https://api.github.com/repos/minio/minio-js/releases

Comment (trim21) [reply github/Requester.py:472]: > Right, `Any` still seems a bit broad to me.

But we can't typed it as `list|dict|None`, if you are using a union type, mypy expect all element of this union match expected types.

For example, `list|dict|None` can't be the argument of  `def f(o: list): ...`

Comment (EnricoMi) [reply github/Requester.py:472]: Awful legacy, well then be it `Any`.

Comment (trim21) [reply github/Requester.py:472]: Not only legacy problem, call dict method on `None|dict` also raise mypy type checking error.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
@trim21 ready to merge?

**Comment by trim21** [comment]
yes

**Comment by trim21** [comment]
I don't have write permission 🙂

**Comment by EnricoMi** [comment]
Thanks!

**Pull request merged**